presc/copies/continuous.py

```diff
@@ -0,0 +1,185 @@
+    incremental training of the classifier copy, which can be carried out
+    continuously non-stop. It keeps taking the necessary data batches from the
+    specified queue as soon as they are available, but it is agnostic as to how
+    the data is added to the queue.
+
+    Attributes
+    ----------
+        classifier_copy : presc.copies.copying.ClassifierCopy
+            An instance of the ClassifierCopy class with the original,
+            classifier, its associated sampling method of choice, and the online
+            copy classifier, which needs to be capable of incremental training.
+        data_stream : queue.Queue
+            The queue where the batches of data for the incremental training of
+            the online copy can be found.
+        fit_kwargs : dict or dict of dicts
+            A dictionary with the fit parameters needed for the online copy
+            classifier, or for each element of the online copy pipeline. In the
+            latter case, there must be an entry for each element of the pipeline
+            and each one in turn must have a dictionary with its fit parameters.
+        verbose : bool
+            If True it provides details at each iteration of the number of
+            samples used so far for the incremental training of the copy, and
+            evaluates the performance of the resulting copy.
+        test_data : presc.dataset.Dataset
+            Dataset with the test data that will be used to evaluate the copy
+            after each training iteration.
+    """
+
+    def __init__(
+        self, classifier_copy, data_stream, fit_kwargs={}, verbose=False, test_data=None
+    ):
+        Thread.__init__(self)
+        self.classifier_copy = classifier_copy
+        self.data_stream = data_stream
+        self.fit_kwargs = fit_kwargs
+        self.verbose = verbose
+        self.test_data = test_data
+
+        self.iterations = 0
+        self.n_samples = 0
+
+        if isinstance(self.classifier_copy.copy, Pipeline):
+            self.is_pipeline = True
+        else:
+            self.is_pipeline = False
+
+    def run(self):
+        """Function to start the copying of the continuous online classifier."""
+        self.continuous_copy_run = True
+
+        self.partial_fit_ok = check_partial_fit(self.classifier_copy.copy)
+        if not self.partial_fit_ok:
+            print("Sorry, not all those transformers/estimators have 'partial_fit'.")
+
+        while self.continuous_copy_run and self.partial_fit_ok:
+            # Get data
+            data_block = self.data_stream.get()
+            n_samples_block = len(data_block.df)
+
+            # If pipeline, train elements of pipeline sequentially,
+            # and use transformed data to fit the next transformer or estimator
+            if self.is_pipeline:
+                n_estimators = len(self.classifier_copy.copy.named_steps)
+                X = data_block.features.copy()
```

Comment:
Possibly not. Or at least I don't see now why it would be.  :thinking: 
In any case, I've run the tests without it and there's no problem so I'll simply remove it.